src/Tools/dotnet-counters/README.md

```diff
@@ -0,0 +1,102 @@
+# dotnet-counters
+
+
+## Install dotnet-counters
+
+```
+ dotnet tool install -g dotnet-counters
+```
+
+
+## Intro
+
+dotnet-counters is a performance monitoring tool for ad-hoc health monitoring or 1st level performance investigation. It can observe performance counter values that are published via `EventCounter` API (https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.tracing.eventcounter). 
+
+
+## Using dotnet-counters
+
+*SYNOPSIS*
+
+    dotnet-counters [--version]
+                    [-h, --help]
+                    <command> [<args>]
+
+*OPTIONS*
+
+    --version
+        Display the version of the dotnet-counters utility.
+
+    -h, --help
+        Show command line help
+
+*COMMANDS*
+
+    list      Display a list of counter names and descriptions
+    monitor   Display periodically refreshing values of selected counters
+
+*LIST*
+
+    dotnet-counters list [-h|--help]
+
+    Display a list of counter names and descriptions, grouped by provider.
+
+    -h, --help
+        Show command line help
+
+    Examples:
+      > dotnet-counters list
+
+    Showing well-known counters only. Specific processes may support additional counters.
+    System.Runtime
+        cpu-usage                    Amount of time the process has utilized the CPU (ms)
+        working-set                  Amount of working set used by the process (MB)
+        gc-heap-size                 Total heap size reported by the GC (MB)
+        gen-0-gc-count               Number of Gen 0 GCs / sec
+        gen-1-gc-count               Number of Gen 1 GCs / sec
+        gen-2-gc-count               Number of Gen 2 GCs / sec
+        exception-count              Number of Exceptions / sec
+
+*MONITOR*
+
+    dotnet-counters monitor [-h||--help]
+                            [-p|--process-id <pid>]
+                            [--refreshInterval <sec>]
+                            counter_list
+    
+    Display periodically refreshing values of selected counters
+    
+    -h, --help
+        Show command line help
+    
+    -p,--process-id
+        The ID of the process that will be monitored
+
+    --refresh-interval
+        The number of seconds to delay between updating the displayed counters
+    
+    counter_list
+        A space separated list of counters. Counters can be specified provider_name[:counter_name]. If the
+        provider_name is used without a qualifying counter_name then all counters will be shown. To discover
+        provider and counter names, use the list command.
+
+    Examples:
```

Comment:
I would put example usage before the syntax reference. Ideally we want someone who is curious to see something working as soon as possible. If the example isn't exactly what they want then the syntax gives them ideas of what else might be supported.

Reply:
It would also be good to have an example of adding a new custom counter in user code.